src/cookbook.js

```diff
@@ -5,11 +5,12 @@ class Cookbook {
   }
 
   findRecipe(searchText) {
+    let filteredRecipes = this.ingredients.filter(x => x.name).filter(x => x.name.includes(searchText)).map(x => x.id)
     return this.recipes.filter(recipe => {
       return recipe.ingredients.find(ingredient => {
-        return (ingredient.name.includes(searchText)) ||
-        (recipe.name.includes(searchText))
-      });x
+        return (filteredRecipes.includes(ingredient.id)) ||
+      (recipe.name.includes(searchText))
+      });
```

Comment:
Kinda hard to read this here, but this now does a few things

1) line 8 filters through the names of the ingredients array (because this is the only place names of ingredients exist) and maps them to a new array by id (because that is how we have to reference them the way the data is set up)

note: the double filter only exists because there is a bud in the api where one name is not present, so it breaks and includes logic.

2. searches the filtered array based on id, also searches based on name for the recipe. The pipe operator servers to return either if, say, garlic is present in  the name or ingredient list.